presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateCountDistinctAggregation.java

```diff
@@ -44,6 +46,16 @@
 
     private ApproximateCountDistinctAggregation() {}
 
+    @InputFunction
+    public static void input(
+            @AggregationState HyperLogLogState state,
+            @BlockPosition @SqlNullable @SqlType("unknown") Block block,
+            @BlockIndex int index,
```

Comment:
Is there any benefit from using `... block, .. index` over `@SqlType("unknown") @SqlNullable Void` as `UnknownOperators` does?

Reply:
We don't support `Void` in the aggregation implementation that reads from `INPUT_CHANNEL` (see `AggregationMetadata.java` and `AccumuloCompiler.java`). This can only be done useing the `BLOCK_INPUT_CHANNEL` or `NULLABLE_BLOCK_INPUT_CHANNEL` approaches.

Reply:
@jessesleeping thanks for the explanation